Index: include/asm-arm/types.h
===================================================================
--- include/asm-arm/types.h.orig	2006-06-11 02:19:22.000000000 -0400
+++ include/asm-arm/types.h	2006-06-11 02:19:25.000000000 -0400
@@ -19,9 +19,9 @@
 typedef __signed__ int __s32;
 typedef unsigned int __u32;
 
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-typedef __signed__ long long __s64;
-typedef unsigned long long __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long __u64;
 #endif
 
 #endif /* __ASSEMBLY__ */
Index: include/asm-i386/types.h
===================================================================
--- include/asm-i386/types.h.orig	2006-06-11 02:19:22.000000000 -0400
+++ include/asm-i386/types.h	2006-06-11 02:19:25.000000000 -0400
@@ -19,9 +19,9 @@
 typedef __signed__ int __s32;
 typedef unsigned int __u32;
 
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-typedef __signed__ long long __s64;
-typedef unsigned long long __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long __u64;
 #endif
 
 #endif /* __ASSEMBLY__ */
Index: include/asm-m32r/types.h
===================================================================
--- include/asm-m32r/types.h.orig	2006-06-11 02:19:22.000000000 -0400
+++ include/asm-m32r/types.h	2006-06-11 02:19:25.000000000 -0400
@@ -23,9 +23,9 @@
 typedef __signed__ int __s32;
 typedef unsigned int __u32;
 
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-typedef __signed__ long long __s64;
-typedef unsigned long long __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long __u64;
 #endif
 #endif /* __ASSEMBLY__ */
 
Index: include/asm-m68k/types.h
===================================================================
--- include/asm-m68k/types.h.orig	2006-06-11 02:19:22.000000000 -0400
+++ include/asm-m68k/types.h	2006-06-11 02:19:25.000000000 -0400
@@ -27,9 +27,9 @@
 typedef __signed__ int __s32;
 typedef unsigned int __u32;
 
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-typedef __signed__ long long __s64;
-typedef unsigned long long __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long __u64;
 #endif
 
 #endif /* __ASSEMBLY__ */
Index: include/asm-mips/types.h
===================================================================
--- include/asm-mips/types.h.orig	2006-06-11 02:19:22.000000000 -0400
+++ include/asm-mips/types.h	2006-06-11 02:19:25.000000000 -0400
@@ -34,9 +34,9 @@
 
 #else
 
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-typedef __signed__ long long __s64;
-typedef unsigned long long __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long __u64;
 #endif
 
 #endif
Index: include/asm-parisc/types.h
===================================================================
--- include/asm-parisc/types.h.orig	2006-06-11 02:19:22.000000000 -0400
+++ include/asm-parisc/types.h	2006-06-11 02:19:25.000000000 -0400
@@ -19,9 +19,9 @@
 typedef __signed__ int __s32;
 typedef unsigned int __u32;
 
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-typedef __signed__ long long __s64;
-typedef unsigned long long __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long __u64;
 #endif
 
 #endif /* __ASSEMBLY__ */
Index: include/asm-sh/types.h
===================================================================
--- include/asm-sh/types.h.orig	2006-06-11 02:19:22.000000000 -0400
+++ include/asm-sh/types.h	2006-06-11 02:19:25.000000000 -0400
@@ -19,9 +19,9 @@
 typedef __signed__ int __s32;
 typedef unsigned int __u32;
 
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-typedef __signed__ long long __s64;
-typedef unsigned long long __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long __u64;
 #endif
 
 #endif /* __ASSEMBLY__ */
Index: include/asm-sparc/types.h
===================================================================
--- include/asm-sparc/types.h.orig	2006-06-11 02:19:25.000000000 -0400
+++ include/asm-sparc/types.h	2006-06-11 02:19:25.000000000 -0400
@@ -28,9 +28,9 @@
 typedef __signed__ int __s32;
 typedef unsigned int __u32;
 
-#ifndef __STRICT_ANSI__
-typedef __signed__ long long __s64;
-typedef unsigned long long __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long __u64;
 #endif
 
 #endif /* __ASSEMBLY__ */
Index: include/asm-x86_64/types.h
===================================================================
--- include/asm-x86_64/types.h.orig	2006-06-11 02:19:22.000000000 -0400
+++ include/asm-x86_64/types.h	2006-06-11 02:19:25.000000000 -0400
@@ -19,8 +19,10 @@
 typedef __signed__ int __s32;
 typedef unsigned int __u32;
 
-typedef __signed__ long long __s64;
-typedef unsigned long long  __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long  __u64;
+#endif
 
 #endif /* __ASSEMBLY__ */
 
Index: include/asm-powerpc/types.h
===================================================================
--- include/asm-powerpc/types.h.orig	2006-06-11 02:19:35.000000000 -0400
+++ include/asm-powerpc/types.h	2006-06-11 02:20:19.000000000 -0400
@@ -40,9 +40,9 @@
 typedef __signed__ long __s64;
 typedef unsigned long __u64;
 #else
-#if defined(__GNUC__) && !defined(__STRICT_ANSI__)
-typedef __signed__ long long __s64;
-typedef unsigned long long __u64;
+#if defined(__GNUC__)
+__extension__ typedef __signed__ long long __s64;
+__extension__ typedef unsigned long long __u64;
 #endif
 #endif /* __powerpc64__ */
 
