--- include.orig/asm-mips/atomic.h	2005-04-21 09:03:16.000000000 +0900
+++ include/asm-mips/atomic.h	2005-05-04 17:02:37.441096029 +0900
@@ -30,6 +30,8 @@ extern spinlock_t atomic_lock;
 
 typedef struct { volatile int counter; } atomic_t;
 
+#ifdef __KERNEL__
+
 #define ATOMIC_INIT(i)    { (i) }
 
 /*
@@ -650,4 +652,6 @@ static __inline__ long atomic64_sub_if_p
 #define smp_mb__before_atomic_inc()	smp_mb()
 #define smp_mb__after_atomic_inc()	smp_mb()
 
+#endif /* __KERNEL__ */
+
 #endif /* _ASM_ATOMIC_H */
--- include.orig/asm-mips/system.h	2005-04-21 09:03:16.000000000 +0900
+++ include/asm-mips/system.h	2005-05-04 17:02:04.181541329 +0900
@@ -12,6 +12,8 @@
 #ifndef _ASM_SYSTEM_H
 #define _ASM_SYSTEM_H
 
+#ifdef __KERNEL__
+
 #include <linux/config.h>
 #include <linux/types.h>
 
@@ -435,4 +437,6 @@ do {						\
 
 #define arch_align_stack(x) (x)
 
+#endif /* __KERNEL__ */
+
 #endif /* _ASM_SYSTEM_H */
