Index: include/asm-mips/atomic.h
===================================================================
--- include/asm-mips/atomic.h.orig	2006-06-05 13:18:23.000000000 -0400
+++ include/asm-mips/atomic.h	2006-06-10 00:22:40.481313069 -0400
@@ -29,6 +29,8 @@
 
 typedef struct { volatile int counter; } atomic_t;
 
+#ifdef __KERNEL__
+
 #define ATOMIC_INIT(i)    { (i) }
 
 /*
@@ -725,5 +727,6 @@
 #define smp_mb__before_atomic_inc()	smp_mb()
 #define smp_mb__after_atomic_inc()	smp_mb()
 
+#endif /* __KERNEL__ */
 #include <asm-generic/atomic.h>
 #endif /* _ASM_ATOMIC_H */
Index: include/asm-mips/system.h
===================================================================
--- include/asm-mips/system.h.orig	2006-06-05 13:18:23.000000000 -0400
+++ include/asm-mips/system.h	2006-06-10 00:22:40.481313069 -0400
@@ -12,5 +12,7 @@
 #ifndef _ASM_SYSTEM_H
 #define _ASM_SYSTEM_H
 
+#ifdef __KERNEL__
+
 #include <linux/types.h>
 
@@ -462,4 +464,6 @@
 
 #define arch_align_stack(x) (x)
 
+#endif /* __KERNEL__ */
+
 #endif /* _ASM_SYSTEM_H */
